starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java

```diff
@@ -1,13 +1,307 @@
+                    } else {
+                        mBox7.setText("O");
+                        mP2Picks.add(7);
+                        mTestingTurnsChange.add(1);
+                        mWhosTurn.setText(mP1Name + "'s turn");
+                    }
+                }else{
+                    Toast.makeText(GameActivity.this, "Please pick a box that has not been picked yet", Toast.LENGTH_SHORT).show();
+                }
+                checkWin();
+            }
+        });
+
+        mBox8.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (mBox8.getText().toString().isEmpty()) {
+                    if (mTestingTurnsChange.size() % 2 == 0) {
+                        mBox8.setText("X");
+                        mP1Picks.add(8);
+                        mTestingTurnsChange.add(1);
+                        mWhosTurn.setText(mP2Name + "'s turn");
+                    } else {
+                        mBox8.setText("O");
+                        mP2Picks.add(8);
+                        mTestingTurnsChange.add(1);
+                        mWhosTurn.setText(mP1Name + "'s turn");
+                    }
+                }else{
+                    Toast.makeText(GameActivity.this, "Please pick a box that has not been picked yet", Toast.LENGTH_SHORT).show();
+                }
+                checkWin();
+            }
+        });
+
+        mBox9.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (mBox9.getText().toString().isEmpty()) {
+                    if (mTestingTurnsChange.size() % 2 == 0) {
+                        mBox9.setText("X");
+                        mP1Picks.add(9);
+                        mTestingTurnsChange.add(1);
+                        mWhosTurn.setText(mP2Name + "'s turn");
+                    } else {
+                        mBox9.setText("O");
+                        mP2Picks.add(9);
+                        mTestingTurnsChange.add(1);
+                        mWhosTurn.setText(mP1Name + "'s turn");
+                    }
+                }else{
+                    Toast.makeText(GameActivity.this, "Please pick a box that has not been picked yet", Toast.LENGTH_SHORT).show();
+                }
+                checkWin();
+            }
+        });
+
+        mBackButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (checkWin() == false){
+                    Toast.makeText(GameActivity.this, "NO GIVING UP...FIGHT TO THE DEATH!!!", Toast.LENGTH_SHORT).show();
+                }else if(mWhosTurn.getText().toString().equals(mP1Name + " WINS!!!")){
+                    Intent test = new Intent(GameActivity.this, MainActivity.class);
+                    test.putExtra("WIN", mP1Name);
+                    startActivity(test);
```

Comment:
The problem with this approach is that you're creating a new instance of MainActivity each time, rather than going back to the existing instance. This stacks up a bunch of instances in memory which can really bog down the user's phone. Instead, we want to `setResult()` and `finish()`.